aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move

```diff
@@ -21,22 +31,34 @@ module aptos_framework::aggregator_v2 {
     public native fun create_aggregator(limit: u128): Aggregator;
 
     /// Adds `value` to aggregator.
-    /// Returns `true` if the addition succeeded and `false` if it exceeded the limit.
+    /// If addition would exceed the limit, `false` is returned, and aggregator value is left unchanged.
     public native fun try_add(aggregator: &mut Aggregator, value: u128): bool;
 
+    // Adds `value` to aggregator, uncoditionally.
+    // If addition would exceed the limit, EAGGREGATOR_OVERFLOW exception will be thrown
+    public fun add(aggregator: &mut Aggregator, value: u128) {
+        assert!(try_add(aggregator, value), error::out_of_range(EAGGREGATOR_OVERFLOW));
+    }
+
     /// Subtracts `value` from aggregator.
-    /// Returns `true` if the subtraction succeeded and `false` if it tried going below 0.
+    /// If subtraction would result in a negative value, `false` is returned, and aggregator value is left unchanged.
     public native fun try_sub(aggregator: &mut Aggregator, value: u128): bool;
 
+    // Adds `value` to aggregator, uncoditionally.
+    // If addition would exceed the limit, EAGGREGATOR_OVERFLOW exception will be thrown
+    public fun sub(aggregator: &mut Aggregator, value: u128) {
+        assert!(try_sub(aggregator, value), error::out_of_range(EAGGREGATOR_UNDERFLOW));
+    }
+
     /// Returns a value stored in this aggregator.
     public native fun read(aggregator: &Aggregator): u128;
 
     public native fun deferred_read(aggregator: &Aggregator): AggregatorSnapshot<u128>;
 
     // Do automatic conversion to u64, if all possible values of aggregator fit it (i.e. limit is <= u64::MAX)
-    // If limit of the aggregator exceeds u64::MAX, this will return None)
-    // This doesn't check if actual value can be converted.
-    public native fun deferred_read_convert_u64(aggregator: &Aggregator): Option<AggregatorSnapshot<u64>>;
+    // If limit of the aggregator exceeds u64::MAX, this will throw an exception.
+    // This doesn't check if actual value can be converted, only if it can always be converted, based on the limit.
+    public native fun deferred_read_u64(aggregator: &Aggregator): AggregatorSnapshot<u64>;
```

Comment:
I am wondering if aggregator is always an int, should the casting be done by having some map/cast API on AggregatorSnapshot directly? So we first deferred read and then map/cast

Reply:
Aggregator snapshot doesn't have a limit any more? And casting of snapshot would suggest it's based on value, instead of the limit